src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -294,9 +294,16 @@ public IActionResult GetAdminToken()
         [Route("admin/warmup")]
         [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
         [RequiresRunningHost]
-        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost)
+        public async Task<IActionResult> Warmup([FromServices] WebJobsScriptHostService hostService)
         {
-            await scriptHost.TryInvokeWarmupAsync();
+            var jobHost = hostService.Services?.GetService<IScriptJobHost>();
```

Comment:
Under what conditions would we be unable to resolve a host here?

Reply:
There is a very small chance we could have a host restart/change of active host to `null` between the time the request is allowed to flow and the time we request the services here. We should be defensive and handle that when it happens as it would lead to a confusing exception otherwise.

Reply:
@mathewc, are you suggesting that we add such null check https://github.com/Azure/azure-functions-host/pull/5108/files/2ce46214dd116dbddb2995475395a78ecbce2fb0#diff-62834549011f3633e270e15bcecc3fe9R306 as well? 
At that point, we probably should make it atomic.
However, Host changing to null is itself a bad state. The error handling just helps with clean exception.

Reply:
Right - what Fabio is saying is what I was getting at. We can't really assume we can get a non-null instance, like we were already assuming in a couple of the existing locations. So unfortunately, callers will have to know to deal with the potential null. An IScriptHostManager.TryGetHost method would help make it clear that there might not be an host presently.